Update python 3.13 priority in 4 packages.

smoser opened the pull request:
These packages did not "just work" with some sed changes to move 3.13 as the highest priority python packages.

py3-build - update python 3.13 version, fix test, improve test 
py3-cairo -  need multiple versions of the py3-cairo-dev 
libmamba - drop python test from base package
py3-google-apitools - do not build 3.13, does not work 

See similar changes #36330

smoser committed:
Update python 3.13 priority in 5 packages.

These packages did not "just work" with some sed changes
to move 3.13 as the highest priority python packages.

py3-build - update python 3.13 version, fix test, improve test
py3-cairo -  need multiple versions of the py3-cairo-dev
libmamba - drop python test from base package
py3-google-apitools - do not build 3.13, does not work

smoser commented:
Note, the package update config check is bogus.  mamba https://github.com/mamba-org/mamba/tags uses semantic versioning, but somehow we got stuck on their YYYY.MM.DD based releases.

xnox commented:
Note @dannf was proposing to downgrade mamba

dannf commented on py3-cairo.yaml:
```diff
@@ -114,23 +127,6 @@ subpackages:
-          with:
-            content: |
-              include="$(python3 -c 'import ${{vars.import}}; print(${{vars.import}}.get_include())')"
-              test -f "$include/py3cairo.h"
```
The reason for this package was to keep the `/usr/include/pycairo/py3cairo.h` file around. I'm not sure if anything uses it, but it could be a regression. The test doesn't seem appropriate though.

smoser replied:
then we should just straight out test for that path. because cairo.get_include() returns `/usr/lib/python3.13/site-packages/cairo/include`

```
$ apk add py3.13-cairo
...
(26/27) Installing cairo (1.18.2-r1)
(27/27) Installing py3.13-cairo (1.27.0-r2)
$  python3.13 -c 'import cairo; print(cairo.get_include())'
/usr/lib/python3.13/site-packages/cairo/include
```

smoser replied:
i think it makes most sense to just drop this 'py3-cairo-dev' package. if something needs it then we can add it. nothing in the archive right now seems to.

dannf commented on py3-cairo.yaml:
```diff
@@ -80,13 +80,26 @@ subpackages:
+        - py3-${{vars.pypi-package}}-dev
+      provider-priority: ${{range.value}}
+    pipeline:
+      - uses: split/dev
```
When I build this locally, these are empty.

smoser replied:
yeah, i guess. and i guess i'm fine to drop this then. 
i'll drop it.
thank you for paying attention.

dannf reviewed (dismissed):
Thanks @smoser ! I highlighted a couple of things w/ pycairo inline.

smoser committed:
drop the py3-cairo-dev package.  it is not used by anything.

At the point where something needs this, then we'll have a use
case for it that we can solve.
